syntax = "proto3";

import "google/api/annotations.proto";

service PetShop {
    // Supplies one kitten to the caller, of the requested breed.
    rpc GetKitten(GetKittenRequest) returns (GetKittenResponse) {
        option (google.api.http) = {
            get: "/v1/kitten/{breed}"
        };
    }
}

enum Breed {
    HALP = 0;
    PERSIAN = 1;
    BENGAL = 2;
    RUSSIAN_BLUE = 3;
    CORNISH_REX = 4;
}

message Kitten {
    string name = 1;
    Breed breed = 2;
    float weight = 3;  // in kg
    int32 age = 4;  // in cat years
}

message GetKittenRequest {
    string breed = 1;
}

message GetKittenResponse {
    Kitten kitten = 1;
}
